app/controllers/templates_controller.rb

```diff
@@ -79,15 +79,21 @@ def auto_complete_controller_name
   end
 
   def preview
-    base = @template.preview_host_collection
-    @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    if params[:id]
+      find_resource
+      base = @template.preview_host_collection
+      @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    else
+      type = request.original_url.split("/")[4]
+      @host = Host.find(params[:preview_host_id])
+      @template = resource_class.new(params[type])
+    end
     if @host.nil?
       render :text => _('No host could be found for rendering the template'), :status => :not_found
       return
     end
-
     @template.template = params[:template]
-    safe_render(@template)
+    safe_render(params[:id] ? @template : @template.delete)
```

Comment:
You understood it correctly, i deleted it in the end because this template wasn't created yet by the user so there is a possibility that the user will do a preview and then close the window without saving, so i use the the object i got after the @template.delete and clearing the record from the db.